agenix.el

```diff
@@ -30,64 +43,67 @@
 
 (defun process-exit-code-and-output (program &rest args)
   "Run PROGRAM with ARGS and return the exit code and output in a list."
-  (with-temp-buffer 
+  (with-temp-buffer
     (list (apply 'call-process program nil (current-buffer) nil args)
           (buffer-string))))
 
 (defun agenix-decrypt-this-buffer ()
   "Decrypt current buffer in a new buffer"
   (interactive)
   (let* ((new-name (concat "*agenix[" (buffer-name) "]*"))
-         (raw-keys (shell-command-to-string "((nix-instantiate --eval -E \"(let rules = import ./secrets.nix; in builtins.concatStringsSep \\\"\n\\\" rules.\\\".netrc.age\\\".publicKeys)\" | sed 's/\"//g' | sed 's/\\\\n/\\n/g') | sed '/^$/d' || exit 1)"))
-         (keys (seq-filter (lambda (s) (not (string= s ""))) (s-split "\n" raw-keys)))
+         (raw-keys (shell-command-to-string
+                    (concat "((nix-instantiate --eval -E \"(let rules = import ./secrets.nix; in builtins.concatStringsSep \\\"\n\\\" rules.\\\""
+                            (buffer-file-name)
```

Comment:
```suggestion
                            (buffer-name)
```
It doesn't work for me without this change. It makes sense, attr name in `secrets.nix` is only a file name, not an absolute path

Reply:
Hunh, you’re right. I wonder how it’s working for me with `(buffer-file-name)`. I just re-loaded and tried it again, and it seems to behave. 🤔 But I’m guessing there’s just some confused state on my end, as I don’t understand how that can work.

Probably `(file-name-nondirectory (buffer-file-name))` is the right thing (for now[^1]), since the buffer could be renamed, or it could have a disambiguated name if two files with the same name in different locations are open.

[^1]: I say “for now”, because I assume (without having dug into agenix) that the .age file could be in some other path relative to secrets.nix, so we might have to search up to find secrets.nix, at which point we want something like `(file-relative-name (buffer-file-name) (file-name-directory secrets-nix-file-name))`.

Reply:
Keep in mind that if you're re-encrypting file without making changes, no actual encryption occurs, so to reproduce you'll need to make a change in decrypted buffer.

I don't know enough about agenix internals, but yeah, current solution won't support nested paths (not sure if they're supported at all by agenix though). `(file-name-nondirectory (buffer-file-name))` is for sure better than `(buffer-name)` because it works even after buffer rename

Reply:
Ah yes, that was exactly it. Thanks. Replicated, fixed, and pushed.